### PR TITLE
Support insecure registry and enhance parsing of image stream tag

### PR DESCRIPTION
**Pull request by qujinping**
2 enhancements for openshift:
1) support insecure docker registry for generation of openshift image stream
we add an import policy for the image stream tag by following instruction of the command argument  
2) enhance parsing of image tag
current implementation doesn't support format like "myregistryhost:5000/fedora/httpd:version1.0". Enhance the parsing logic.

### CODE REVIEW

**Comment by pradeepto** [comment]
Hey @qujinping , Thank you for your patch. This is really nice, we will look at it. But can you please sign the CLA first or else this PR is blocked.

**Comment by qujinping** [comment]
I signed it

**Comment by kadel** [comment]
@qujinping It look like k8s-ci-robot  is not picking that up :-(
Have you used same email as you have in git commit?

Or you could try push something to your branch so it forces bot to recheck.

File: cmd/up.go [hunk 1]
```diff
@@ -54,6 +56,7 @@ var upCmd = &cobra.Command{
 }
 
 func init() {
+	upCmd.Flags().BoolVar(&UpInsecureRepo, "insecure-repository", false, "Use Insecure Repository")
```
Comment (surajssd) [inline cmd/up.go]: if we are planning to add this as a flag then we should do it as a global flag!

Comment (jjqquu) [inline cmd/up.go]: @surajssd , any suggestion for me to improve?

Right now, it only impacts convertion of image stream for openshift.

Comment (kadel) [inline cmd/up.go]: @surajnarwade why as global? 
This flag make sense only for `up` and `convert` and only for openshift.

Comment (kadel) [inline cmd/up.go]: we will have to clean up flags a bit, so they are defined in one place, but that should be solved in different PR

**Comment by kadel** [comment]
ping @cdrage @surajssd

**Comment by cdrage** [comment]
@qujinping Almost there! Thanks a lot for this contribution. The code LGTM.

One thing that needs to be adding is here: https://github.com/qujinping/kompose/blob/e4302af6fe3fdf7eb9944438cd13022f2e62f2a5/cmd/convert.go#L147

Since we have a custom output for `--help` an entry for Insecure Registry should be there. 

I also have another comment in regards to the description which I'll put in a code review.

File: cmd/convert.go [hunk 2]
```diff
@@ -103,6 +105,7 @@ func init() {
 
 	// OpenShift only
 	convertCmd.Flags().BoolVar(&ConvertDeploymentConfig, "deployment-config", true, "Generate an OpenShift deploymentconfig object")
+	convertCmd.Flags().BoolVar(&ConvertInsecureRepo, "insecure-repository", false, "Use Insecure Repository")
```
Comment (cdrage) [inline cmd/convert.go]: No need for capitalization! Correct grammar is `Use an insecure repository`.

It might be better to add a note that this is OpenShift only for now. So perhaps `Use an insecure OpenShift repository`, would be better.

**Comment by qujinping** [comment]
@cdrage, thanks for your comment, accepted. 

Pls review my update.

**Comment by qujinping** [comment]
I close this PR because I messed things up while merging from upstream.

A new PR will be created instead. Very sorry for the inconvenience.
